tests/stdlib/time/instant.dora

```diff
@@ -0,0 +1,6 @@
+fun main() {
+  /* fail: "expected class" */
+  // let x = std::time::Instant(0L, 0);
+  // let y = std::time::Instant::EPOCH();
+  // assert(x == y);
```

Comment:
There are quite some tests for namespaces, so you can see what's working from there. Should be easy to simplify this code here to a few lines and run/test/debug it.

Reply:
thanks, I'll check again, couldn't find examples last time

Reply:
Ok, I checked and I haven't found any additional examples in *.dora or as tests in *.rust files.

I couldn't make any combination of `import std::time::Instant;`, `namespace` and qualifying the `Instant` type work. Can you explain in a sentence how it's supposed to work, so I can dig in and figure out why it doesn't?

Reply:
Tests are here for example: https://github.com/dinfuehr/dora/blob/master/dora/src/language/fctbodyck/tests.rs#L2980

Reply:
Yes, these are the tests I have been aware of.

This doesn't work:

```
import std::time::Instant;
fun main() { Instant(0L, 0); }
// instant.dora at 1:1: path expected.
```

Maybe we don't need `std`?

```
import time::Instant;
fun main() { Instant(0L, 0); }
// instant.dora at 1:1: path expected.
```

Nope. Maybe without the type itself?

```
import std::time;
fun main() { time::Instant(0L, 0); }
// instant.dora at 1:1: namespace `std` does not have field named `time`.
```


All of the above with `Instant` enclosed in `namespace time`:

```
import std::time::Instant;
fun main() { Instant(0L, 0); }
// instant.dora at 1:1: path expected.
```

```
import time::Instant;
fun main() { Instant(0L, 0); }
// instant.dora at 1:1: path expected.
```

```
import std::time;
fun main() { time::Instant(0L, 0); }
// instant.dora at 1:1: namespace `std` does not have field named `time`.
```

Now with `@pub namespace time`:

```
import std::time::Instant;
fun main() { Instant(0L, 0); }
// instant.dora at 1:1: path expected.
```

```
import time::Instant;
fun main() { Instant(0L, 0); }
// instant.dora at 1:1: path expected.
```

```
import std::time;
fun main() { time::Instant(0L, 0); }
// instant.dora at 1:1: namespace `std` does not have field named `time`.
```

Reply:
¯\_(ツ)_/¯

Reply:
I mean that you construct the same program as a small test:

```
namespace mystd { namespace time { class Instant... } }
fun f() { mystd::time::Instant() }
```

We definitely need a better error message here than `path expected`. I suspect it's because of sub-directories in the stdlib. You can try to move the Instant class out of that time directory.

Reply:
Ok, then I'll investigate the sub-dir issue, I think writing `namespace mystd { namespace time { class Instant... } }` is not something that should be seriously considered outside test snippets.

Reply:
@dinfuehr Could you have a look at #205 and #279 while you're at it?

Reply:
Btw I warned about this behavior in this [PR](https://github.com/dinfuehr/dora/pull/241).

Reply:
> Ok, then I'll investigate the sub-dir issue, I think writing `namespace mystd { namespace time { class Instant... } }` is not something that should be seriously considered outside test snippets.

Well, obviously. It is perfect to test though what's failing here and if you look at the test I wrote above, I am pretty sure that there already exists a similar looking test. So it helps us to figure out that something is wrong with sub-directories here in the stdlib.

As I have pointed out already, this bug is due to one of your recent changes that just puts all stdlib files into one namespace. Namespaces are broken due to this in the stdlib. Your code should work fine in user-code. This is specifically happening with the stdlib since its content is loaded from a global in the binary instead of from disk and uses different code paths due to that.

That said, I am not going to merge this PR anytime soon. I don't want another unfinished feature in Dora, especially since this code doesn't do anything useful at all at the moment. I would rather polish and improve the stuff we already have. Here this would mean fixing namespaces in the stdlib.